Import file path memorization

sgourdas opened the pull request:
Added functionality for the download path to be retrievable and modifiable. On the settings menu, based on the discussion on #976, there is now a new section for the import directory to be set. With this, after a restart, the import directory stays the same based on what the user has selected previously and is saved in settings

Added:
- Logic to keep last location of import remembered
- Option in settings for default import directory

This closes #976

kelson42 commented:
@sgourdas Would you be able please to rebase (and fix conflict) based on `main` so we (finally) can move forward with the review?

veloman-yunkan reviewed (commented):
I don't think that we have to have an explicit entry in settings for the import dir. Rather it should be implicit information saved in the session file similar to how its done for the list of open tabs (see #1048).

sgourdas commented:
@veloman-yunkan I feel like in Kiwix, opening a file is a more common every day task, so it would be better if we clearly show the path in settings. I assume that is also why we have the `Open file` in the mainwindow and not in the main menu dropdown, because it is part of the main workflow. 

In any case I can understand why it could also not be thought as ideal. If you agree, I can go ahead and transfer the logic to the session file.

veloman-yunkan commented:
@sgourdas It's up to @kelson42 to decide

kelson42 commented:
@sgourdas @veloman-yunkan Sorry for the late feedback. We have to stick to the issue. This does not imply any new setting. We just need to remember over restart the last path used to load manually a ZIM file... and if not set the use the "Download" directory pre-configured by the OS.

kelson42 commented:
@sgourdas Hi, do you still plan to update/complete this PR?

sgourdas commented:
Hi @kelson42,

Sorry for the lack of feedback, I am out of base until Wednesday.

sgourdas commented:
@kelson42 I reverted the code surrounding the setting logic and implemented the memorization in the session file.

Also, fixed a small typo `Splitted` to `Split`

kelson42 reviewed (changes requested):
Thank you for the update, here my remarks:
* Please secure that if the settings "import/dir" is not set you fallback to the OS "Download" directory and otherwise fallback to the current directory
* Settings naming is not optimal IMHO, something around "default-third-party-zim-directory-path" (sorry, not aware about the settings naming nomenclature) would be clearer
* Please be careful not to add trailing spaces/tabs. You can can the following in your `.gitconfig`
```
[core]
        whitespace = trailing-space,space-before-tab,tab-in-indent
```

@sgourdas Be careful, I have rebased on git `main HEAD` your branch.

sgourdas commented:
@kelson42 got it. For the naming, would it be ok if we use something smaller like just `zim-import-dir`?

kelson42 commented:
@sgourdas why not ;)

sgourdas commented:
@kelson42 I am not sure if the current state of the commits is ok based on the rebase with main :') Please advise me

kelson42 commented:
@sgourdas please squash all commits and rebase on main.

kelson42 commented:
@sgourdas Don‘t worry to mich about the Windows CI, but failing Linux CI seems to be a blocker!

sgourdas commented:
Sorry for the total mess @kelson42. I think we are ok now, hopefully you agree.

kelson42 reviewed (approved):
@sgourdas LGTM, Thx.
@veloman-yunkan Can you please do the code review?

veloman-yunkan reviewed (approved):
LGTM
